### ISSUE TITLE
FloatingFocusManager adding aria-hidden to Dropdown trigger

### ISSUE THREAD

**Issue by anuraghazra** [razorpay/blade#1686]
Reference PR: https://github.com/razorpay/blade/pull/1489

--- 

Context: https://razorpay.slack.com/archives/CMQ3RBHEU/p1695972277506429?thread_ts=1693739607.070049&cid=CMQ3RBHEU

> In [this](https://github.com/razorpay/blade/pull/1489) PR we added the "[FloatingFocusManager](https://floating-ui.com/docs/floatingfocusmanager)" from FloatingUI to automate the focus management logic and added the modal=true prop to it, this [modal](https://floating-ui.com/docs/floatingfocusmanager#modal) prop will ensure that everything outside of the FloatingFocusManager will get aria-hidden which simulates the [inert](https://github.com/WICG/inert) prop so that focus is trapped within the modal. 
Functionally the Dropdown worked as expected but UTs failed because FloatingFocusManager added aria-hidden even to the Dropdown trigger.
